### PR TITLE
[Compute] Add new license_type in vm create/update: RHEL_BYOS, SLES_BYOS

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

Resolve https://github.com/Azure/azure-cli/issues/14735
[Compute] Add new license_type in vm create/update: RHEL_BYOS, SLES_BYOS

**Testing Guide**  

```
az vm create --license-type RHEL_BYOS
az vm update --license-type RHEL_BYOS
```

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -811,7 +811,7 @@ def load_arguments(self, _):
                           "To enable Azure Hybrid Benefit for Windows Server, use 'Windows_Server'. " \
                           "To enable Multitenant Hosting Rights for Windows 10, use 'Windows_Client'. " \
                           "For more information see the Azure Windows VM online docs."
-            c.argument('license_type', help=license_msg, arg_type=get_enum_type(['Windows_Server', 'Windows_Client', 'None']))
+            c.argument('license_type', help=license_msg, arg_type=get_enum_type(['Windows_Server', 'Windows_Client', 'RHEL_BYOS', 'SLES_BYOS', 'None']))
```
Comment (arrownj) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:814]: No enum defined in  SDK ?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:814]: No. It is a string type in swagger.

**Comment by yonzhan** [comment]
Compute

**Comment by yungezz** [comment]
This is for a Ignite feature, but may not catch Ignite schedule, because service side Swagger merged at end of Aug, and deploy after that. Service team just need a offline package for Ignite demo purpose, not official release.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 1 above)
Comment (yungezz) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:814]: some general question.
- for this kind of string type enum in Swagger, how about file an issue in swagger repo to service team to get awareness?
- case insenstive/sensitive rule at CLI side?  Or we just pass in what service required?

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:814]: 1. They know it. They don't declare they will fix it. I am OK with string type.
2. Enum type in CLI supports case insensitive matching.

**Comment by asinn826** [comment]
@yungezz - do you know when this will roll out in the CLI? If it's after Ignite we will have to tweak our Ignite messaging on the marketing side.

**Comment by yungezz** [comment]
hi @asinn826 it depends on when service side is ready, if service side could ready now, we can merge the PR immediately, so it will catch azure CLI Ignite release

**Comment by asinn826** [comment]
@yungezz - CRP's changes have been made and are rolling out right now - currently in Canary. Is that sufficient?

**Comment by yungezz** [comment]
> @yungezz - CRP's changes have been made and are rolling out right now - currently in Canary. Is that sufficient?

When will it be on Prod? Azure CLI releasing is from user perspective bar. If user could access the feature with Azure CLI now, then we can do the release.

**Comment by qwordy** [comment]
The next release day is 09/22/2020

**Comment by asinn826** [comment]
> The next release day is 09/22/2020

Thanks @qwordy - does that mean the next CLI version containing the license type updates will be released and available for customers on 9/22/2020? If so, that would align perfectly with platform rollouts and expected announcements.

**Comment by qwordy** [comment]
Yes. If the feature is available for users, we can release it.

**Review by arrownj — APPROVED** [review]

**Pull request merged**